pallets/dactr/src/lib.rs

```diff
@@ -130,7 +132,21 @@ pub mod pallet {
 		) -> DispatchResultWithPostInfo {
 			let who = ensure_signed(origin)?;
 			ensure!(!data.is_empty(), Error::<T>::DataCannotBeEmpty);
-			Self::deposit_event(Event::DataSubmitted { who, data });
+
+			// SAFETY: `data.len()` is always less than `u32::MAX` because it is bounded by
+			// `BoundedVec`
+			let len = data.len() as u32;
```

Comment:
`as u32` never panics and therefore safe. You should try:
```rust
let len = u32::try_from(data.len()).expect("`data` has bounded length therefore < u32::MAX");
```

Reply:
Hi @i1i1 
As you know in some platforms `usize >> u32`, and that is why we usually use `u32::try_from`. However, as: 
- `data` is a `BoundedVec<T,S>` where `S: Get<u32>` and
- `Config::MaxAppDataLenght` is also `Get<u32>`, 
we are sure that `data.len` can be safely converted to `u32` without extra checks at runtime.